### PR TITLE
fix: link for resolved comment does not show a comment thread

### LINKED ISSUE #5434: [Bug] The post button is not displayed when the user is adding a new comment thread on Macbook
## Description

[What happened]specific Mac ->collab->the post button is not displayed when the user is adding a new comment thread

### Steps to reproduce the behaviour:

1. click on the comment icon
2. drop the comment icon
3. add the the first comment thread
4. verify the display of post button

image

### Important Details

- Version: [Cloud]
- OS: Mac
- Browser Safari
- Environment release

### LINKED ISSUE #5811: [Bug] Link for a resolved comment thread notification does not show the comment
## Description
Link for a resolved comment thread notification does not show the comment. Since the `show resolved comments` filter is not set

1. Create an app
2. Invite a new user
3. Add a comment and tag the new user
4. Resolve the comment
5. Login as the invited user and click on the comment notification
6. The thread is not shown since the resolved filter is not set

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
When user tries to open a link of resolved comment thread, we'll enforce the `show resolved comments` filter

Fixes #5811 
Fixes #5434 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/5811-link-for-resolved-thread-not-visible 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.92 **(0)** | 36.82 **(0)** | 33.64 **(-0.01)** | 55.46 **(-0.01)**
 :green_circle: | app/client/src/comments/inlineComments/Comments.tsx | 97.22 **(0.08)** | 90 **(0)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/components/designSystems/appsmith/CenteredWrapper.tsx | 66.67 **(-33.33)** | 100 **(0)** | 0 **(-100)** | 66.67 **(-33.33)**
 :red_circle: | app/client/src/constants/messages.ts | 70.84 **(-0.02)** | 100 **(0)** | 12.03 **(-0.04)** | 78.43 **(0.06)**</details>

### CODE REVIEW

File: app/client/src/comments/AppComments/AppCommentThreads.tsx [hunk 1]
```diff
@@ -26,6 +28,8 @@ const Container = styled.div`
 `;
 
 function AppCommentThreads() {
+  const dispatch = useDispatch();
+  const commentThreadIdInUrl = useSelectCommentThreadUsingQuery();
```
Comment (sbalaji1192) [inline app/client/src/comments/AppComments/AppCommentThreads.tsx:32]: `commentThreadIdFromUrl` seems like better name. `commentThreadIdInUrl` sounds like a boolean value.

**Review by sbalaji1192 — APPROVED** [review]
LGTM

**Review by sbalaji1192 — APPROVED** [review]
LGTM

File: app/client/src/comments/AppComments/AppCommentThreads.tsx [hunk 2]
```diff
@@ -56,6 +60,17 @@ function AppCommentThreads() {
     ],
   );
 
+  useEffect(() => {
+    // if user is visiting a comment thread link which is already resolved,
+    // we'll activate the resolved comments filter
+    if (
+      commentThreadIdFromUrl &&
+      !commentThreadIds.includes(commentThreadIdFromUrl)
```
Comment (rishabhsaxena) [inline app/client/src/comments/AppComments/AppCommentThreads.tsx:68]: nit: a thread can also be deleted, so maybe we should check if the thread has a active resolved state

**Review by rishabhsaxena — DISMISSED** [review]

File: app/client/src/comments/AppComments/AppCommentThreads.tsx (hunk 2 above)
Comment (pranavkanade) [reply app/client/src/comments/AppComments/AppCommentThreads.tsx:68]: In case of deleted msg we don't show user any msg, shall I add a toast?

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**